extensions/default/src/DataSourceConfigurationAPI/GoogleCloudDataSourceConfigurationAPI.ts

```diff
@@ -0,0 +1,192 @@
+import { ExtensionManager, Types } from '@ohif/core';
+
+/**
+ * The various Google Cloud Healthcare path item types.
+ */
+enum ItemType {
+  projects = 0,
+  locations = 1,
+  datasets = 2,
+  dicomStores = 3,
+}
+
+interface NamedItem {
+  name: string;
+}
+interface Project extends NamedItem {
+  projectId: string;
+}
+
+/**
+ * Fetches an array of items the specified item type.
+ * @param urlStr the fetch url
+ * @param fetchItemType the type to fetch
+ * @param fetchOptions the header options for the fetch (e.g. authorization header)
+ * @param fetchSearchParams any search query params; currently only used for paging results
+ * @returns an array of items of the specified type
+ */
+async function _doFetch(
+  urlStr: string,
+  fetchItemType: ItemType,
+  fetchOptions = {},
+  fetchSearchParams: Record<string, string> = {}
+): Promise<Array<Project> | Array<NamedItem>> {
+  try {
+    const url = new URL(urlStr);
+    url.search = new URLSearchParams(fetchSearchParams).toString();
+
+    const response = await fetch(url, fetchOptions);
+    const data = await response.json();
+    if (response.status >= 200 && response.status < 300 && data != null) {
+      if (data.nextPageToken != null) {
+        fetchSearchParams.pageToken = data.nextPageToken;
+        const subPageData = await _doFetch(
+          urlStr,
+          fetchItemType,
+          fetchOptions,
+          fetchSearchParams
+        );
+        data[ItemType[fetchItemType]] = data[ItemType[fetchItemType]].concat(
+          subPageData
+        );
+      }
+      return data[ItemType[fetchItemType]];
+    } else {
+      const message =
+        data?.error?.message ||
+        `Error returned from Google Cloud Healthcare: ${response.status} - ${response.statusText}`;
+      throw new Error(message);
+    }
+  } catch (err) {
+    const message = err?.message || 'Error occurred during fetch request.';
+    throw new Error(message);
+  }
+}
+
+const initialUrl = 'https://cloudresourcemanager.googleapis.com/v1';
```

Comment:
shouldn't these come from the datasource config?

Reply:
They can, but for a Google data source I believe these will be constant in general.